waku/waku.go

```diff
@@ -145,25 +145,25 @@ func New(cfg *Config, logger *zap.Logger) *Waku {
 		LightClient:              cfg.LightClient,
 		BloomFilterMode:          cfg.BloomFilterMode,
 		RestrictLightClientsConn: cfg.RestrictLightClientsConn,
-		SyncAllowance:            DefaultSyncAllowance,
+		SyncAllowance:            common.DefaultSyncAllowance,
 	}
 
 	if cfg.FullNode {
-		waku.settings.BloomFilter = MakeFullNodeBloom()
-		waku.settings.BloomFilterTolerance = MakeFullNodeBloom()
+		waku.settings.BloomFilter = common.MakeFullNodeBloom()
+		waku.settings.BloomFilterTolerance = common.MakeFullNodeBloom()
 	}
 
-	waku.filters = NewFilters(waku)
+	waku.filters = common.NewFilters()
 
 	// p2p waku sub-protocol handler
 	waku.protocol = p2p.Protocol{
-		Name:    ProtocolName,
-		Version: uint(ProtocolVersion),
-		Length:  NumberOfMessageCodes,
+		Name:    v0.Name,
+		Version: uint(v0.Version),
+		Length:  v0.NumberOfMessageCodes,
 		Run:     waku.HandlePeer,
 		NodeInfo: func() interface{} {
 			return map[string]interface{}{
-				"version":        ProtocolVersionStr,
+				"version":        v0.VersionStr,
 				"maxMessageSize": waku.MaxMessageSize(),
 				"minimumPoW":     waku.MinPow(),
```

Comment:
I imagine this is the place where all the supported versions are declared? I think I remember from your draft that this was an array of `p2p.Protocol`, this could be the main section that any potential `PeerFactory{}` uses to determine which peer to instantiate.